.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,15 @@
+name: Run pre-commit
+
+on:
+  workflow_call:
+
+jobs:
+  run-pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+      - run: python3 -m pip install pre-commit
```

Comment:
We could also cache pre-commit venvs for this. remember the caches are specific to the repository that called the reusable workflow.

Reply:
```yml
      - uses: actions/setup-python@v5
        id: python-setup
        with:
          python-version: '3.x'
      - name: Cache pre-commit environments
        uses: actions/cache@v4
        with:
          path: '~/.cache/pre-commit'
          key: pre-commit-${{ steps.python-setup.outputs.python-version }}-${{ hashFiles('.pre-commit-config.yaml') }}
      - run: pipx run pre-commit run --all-files
```